File: GCD-eulcidean-algorithm.py
def compute_gcd(a, b):
    while b != 0:
        a, b = b, a % b
    return a

def main():
    try:
        
        A = int(input("Enter the first integer (A): "))
        B = int(input("Enter the second integer (B): "))

        
        gcd = compute_gcd(A, B)

       
        print(f"The GCD of {A} and {B} is: {gcd}")

    except ValueError:
        print("Please enter valid integers.")


if __name__ == "__main__":
    main()

File: GCD-eulcidean-algorithm.c
#include <stdio.h>

int compute_gcd(int a, int b) {
    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int main() {
    int A, B;

  
    printf("Enter the first integer (A): ");
    scanf("%d", &A);
    printf("Enter the second integer (B): ");
    scanf("%d", &B);

    int gcd = compute_gcd(A, B);

    
    printf("The GCD of %d and %d is: %d\n", A, B, gcd);

    return 0;
}